site/app/libraries/database/DatabaseQueries.php

```diff
@@ -1100,18 +1110,46 @@ public function getParentPostId($child_id) {
         return $this->course_db->rows()[0]['parent_id'];
     }
 
-    public function editPost($original_creator, $user, $post_id, $content, $anon, $markdown) {
+    /**
+     * @return string[]
+     */
+    public function getForumAttachments(int $post_id, int $version = 0) {
+        // Default version is current version
+        if ($version === 0) {
+            $version = $this->getPost($post_id)['version_id'];
+        }
+        $this->course_db->query("SELECT file_name FROM forum_attachments WHERE post_id = ? AND version_added <= ? AND (version_deleted > ? OR version_deleted = 0)", [$post_id, $version, $version]);
+        // Strip 'file_name' keys from each row
+        $attachment_name = [];
+        foreach ($this->course_db->rows() as $row) {
+            $attachment_name[] = $row['file_name'];
+        }
+        return $attachment_name;
+    }
+
+    /**
+     * @param string[] $attachments_deleted
+     */
+    public function editPost($original_creator, $user, $post_id, $content, $anon, $markdown, $attachments_deleted) {
         try {
             $markdown = $markdown ? 1 : 0;
             // Before making any edit to $post_id, forum_posts_history will not have any corresponding entry
             // forum_posts_history will store all history state of the post(if edited at any point of time)
             $this->course_db->beginTransaction();
             // Insert first version of post during first edit
-            $this->course_db->query("INSERT INTO forum_posts_history(post_id, edit_author, content, edit_timestamp) SELECT id, author_user_id, content, timestamp FROM posts WHERE id = ? AND NOT EXISTS (SELECT 1 FROM forum_posts_history WHERE post_id = ?)", [$post_id, $post_id]);
+            $this->course_db->query("INSERT INTO forum_posts_history(post_id, edit_author, content, edit_timestamp, has_attachment, version_id) SELECT id, author_user_id, content, timestamp, has_attachment, version_id FROM posts WHERE id = ? AND NOT EXISTS (SELECT 1 FROM forum_posts_history WHERE post_id = ?)", [$post_id, $post_id]);
+            // Get latest version number
+            $this->course_db->query("SELECT version_id FROM posts WHERE id = ?", [$post_id]);
+            $version_id = $this->course_db->rows()[0]["version_id"] + 1;
+            // Update attachments
+            foreach ($attachments_deleted as $img) {
+                $this->course_db->query("UPDATE forum_attachments SET version_deleted = ? WHERE post_id = ? AND file_name = ?", [$version_id, $post_id, $img]);
```

Comment:
Same comment here.  You can turn this into a multiple-update instead of a loop of updates.